### ISSUE TITLE
Which Build Threat corresponds to "pwn request"

### ISSUE THREAD

**Issue by fproulx-boostsecurity** [slsa-framework/slsa#1235]
I'm trying to fit classic "pwn request" in an existing Build Threat (https://slsa.dev/spec/v1.0/threats#e-compromise-build-process) and it's not 100% clear in which one it neatly fits in.

It clearly falls under `(E) Compromise build process`, but the detailed threat scenarios do not really cover the classic attack sequence: Public GitHub Repo, Repo Uses GitHub Actions, Threat Actor forks repo, Workflow triggers on pull_request_target event, Workflow checks out untrusted code from fork, Workflow has either Injection or uses a compiler / linter / test tool that is known to consume source files in a way that can influence build behavior (i.e.  "Living Off The Pipeline")

In the 1.1 (or Draft), I see `(D) External build parameters` which has `Build from unofficial parameters` which mentions "injection", but that still does not talk about Pull Requests from forks, nor does not cover the "execution influenced via untrusted code checkout"

**Comment by fproulx-boostsecurity** [comment]
I could find a discussion between @MarkLodato and @mattmoor that mentioned `pull_request` / `pull_request_target` (https://github.com/slsa-framework/slsa/issues/188) and that was put on the back burner it seems

**Comment by mattmoor** [comment]
I think the key thing is that `pull_request_target` can still be misused to activate unreviewed code, so it isn't a silver bullet.

At you are well aware, this is unfortunately extremely nuanced (to the point that I basically wholly discourage the use of `pull_request_target` because it's effectively impossible to "hold safely").

**Comment by fproulx-boostsecurity** [comment]
It is possible to use it safely, but it comes with great responsibility, break up of workflow into multiple jobs isolating the dangerous parts, making it have zero access (least privilege setting of GITHUB TOKEN for each job,  no secret ), pipe and filter of risky job output to next stage etc. 

In any case, SLSA should clearly document those threat because it’s still very much prevalent and there are much less obvious scenarios

**Comment by mattmoor** [comment]
“Effectively” is very load bearing in my statement.

The level of awareness is exceptionally (I’d say prohibitively) high

**Comment by fproulx-boostsecurity** [comment]
The fact remains that the same idea applies across to Gitlab pipelines, Azure Devops, Circleci , Openshift they all have this privileged mode which is risky.

**Comment by trishankatdatadog** [comment]
Looks like some of the [TODOs for mitigating some attacks in v1.1](https://github.com/slsa-framework/slsa/issues?q=is%3Aopen+is%3Aissue+label%3A%22slsa+1.1%22).

I just don't think we're there yet to be able to say anything precise enough for this. The way I see it, SLSA is about using in-toto attestations to concretely address some attacks. We only prescribe one attestation today (build provenance), and how can it be used to address this attack? Also, like Matt says, the mitigation is nuanced (and specific to GitHub anyway).

**Comment by fproulx-boostsecurity** [comment]
I understand that the scope of SLSA on a technical level is attestation and build provenance, but the fact remains that to this day it's the best trustworthy reference generic Threat Model for the Software Supply Chain Security of an average SDLC.

I agree with Matt that it's hard to do well and it's any guidance is highly implementation and platform specific, but the general concept of "pwn request" in the context of Open Source projects hosted on public source control (not just GitHub) and that accept contributions and trigger linting / tests on the contribution is not specific. 

GitHub / GitLab / Azure DevOps / BitBucket, etc. all have the concept of forks and can trigger their own built-in Build (or external, as a side effect).

I go back to my original question, does that fit in `Compromise build process` or `External build parameters`. I am certainly NOT asking for SLSA to have tons platform specific gotchas. Instead I'm just looking to capture to generic threat an agnostic manner.

Then there are plenty of resources (shameless plug to my own) that deal with implementation specific gotchas.

**Comment by trishankatdatadog** [comment]
> Returning to my original question: does this scenario fit under "Compromise build process" or "External build parameters"? I'm certainly not suggesting that SLSA should include an extensive list of platform-specific nuances. Instead, I’m simply looking to capture this generic threat in an agnostic manner.

Great points, and make sense. To me, I think "pwn request" falls more under "External build parameters". WDYT?

**Comment by fproulx-boostsecurity** [comment]
Yes. I’ve we think of “injection” that’s clear, if we think of untrusted code from the head of the fork as external build “parameter” bit of a stretch but it’s an input fed to say “npm install” leading to RCE

**Comment by fproulx-boostsecurity** [comment]
With all the TeamPCP stuff maybe it’s time to document this better as a threat in SLSA. Especially when you have people making in incorrect claims like https://x.com/cloudsa/status/2058201590365147298?s=46&t=aqRCUtFE8OnUPSWxEHt-PQ
